exporter.go

```diff
@@ -301,56 +311,84 @@ func (l *StatsDListener) handlePacket(packet []byte, e chan<- Events) {
 			continue
 		}
 
-		elements := strings.Split(line, ":")
+		elements := strings.SplitN(line, ":", 2)
 		if len(elements) < 2 {
 			networkStats.WithLabelValues("malformed_line").Inc()
 			log.Println("Bad line from StatsD:", line)
 			continue
 		}
 		metric := elements[0]
-		samples := elements[1:]
+		var samples []string
+		if strings.Contains(elements[1], "|#") {
+			// using datadog extensions, disable multi-metrics
+			samples = elements[1:]
+		} else {
+			samples = strings.Split(elements[1], ":")
+		}
 		for _, sample := range samples {
 			components := strings.Split(sample, "|")
 			samplingFactor := 1.0
-			if len(components) < 2 || len(components) > 3 {
+			if len(components) < 2 || len(components) > 4 {
 				networkStats.WithLabelValues("malformed_component").Inc()
 				log.Println("Bad component on line:", line)
 				continue
 			}
 			valueStr, statType := components[0], components[1]
+			labels := map[string]string{}
```

Comment:
Let's move this down above the len(components) check to keep related code together.